system variables: add two TSO related variables (#6770)
pingcap/docs: system-variables.md

- Default value: `ON`
- This variable is used to dynamically control whether the telemetry collection in TiDB is enabled. By setting the value to `OFF`, the telemetry collection is disabled. If the [`enable-telemetry`](/tidb-configuration-file.md#enable-telemetry-new-in-v402) TiDB configuration item is set to `false` on all TiDB instances, the telemetry collection is always disabled and this system variable will not take effect. See [Telemetry](/telemetry.md) for details.

### tidb_enable_tso_follower_proxy <span class="version-mark">New in v5.3</span>

- Scope: GLOBAL
- Default value: `OFF`

> - This variable is used to enable the TSO Follower Proxy feature. When the value is `OFF`, TiDB only gets TSO from PD leaders. After this feature is enabled, TiDB gets TSO by evenly sending requests to all PD nodes and forwarding TSO requests through PD followers. By that means, the CPU pressure of PD leaders is reduced.

Reviewer: ```suggestion
- This variable is used to enable the TSO Follower Proxy feature. When the value is `OFF`, TiDB only gets TSO from the PD leader. After this feature is enabled, TiDB gets TSO by evenly sending requests to all PD nodes and forwarding TSO requests through PD followers. This helps reduce the CPU pressure of PD leader.
```

- Scenarios for enabling TSO Follower Proxy:

>     * Due to the high pressure of TSO requests, the CPU of PD leaders reaches a bottleneck, which causes high latency of TSO RPC requests.

Reviewer: ```suggestion
    * Due to the high pressure of TSO requests, the CPU of the PD leader reaches a bottleneck, which causes high latency of TSO RPC requests.
```

>     * Due to the high pressure of TSO requests, the CPU of PD leaders reaches a bottleneck, and this issue causes high latency of TSO RPC requests.

Reviewer 2: ```suggestion
    * Due to the high pressure of TSO requests, the CPU of PD leaders reaches a bottleneck, which causes high latency of TSO RPC requests.
```

>     * The TiDB cluster has a large number of TiDB instances, and increasing the value of [tidb_tso_client_batch_max_wait_time](#tidb_tso_client_batch_max_wait_time-new-in-v53) cannot alleviate the issue of high latency of TSO RPC requests.

Reviewer: ```suggestion
    * The TiDB cluster has many TiDB instances, and increasing the value of [`tidb_tso_client_batch_max_wait_time`](#tidb_tso_client_batch_max_wait_time-new-in-v53) cannot alleviate the high latency issue of TSO RPC requests.
```

> **Note:**
>

> > Suppose the TSO RPC latency increases, but the CPU bottleneck of PD leaders does not cause this issue. In this case, enabling the TSO Follower Proxy might increase the latency of TiDB statement execution, thereby affecting the QPS performance of the cluster.

Reviewer: ```suggestion
> Suppose that the CPU of the PD leader does not reach a bottleneck and the TSO RPC latency does not increase. In this case, enabling the TSO Follower Proxy might increase the execution latency in TiDB and affect the QPS performance of the cluster.
```

Author: 这部分的第一句改成这样如何呢？
Suppose that the CPU of the PD leader does not reach a bottleneck but the TSO RPC latency increases.

Reviewer: @JmPotato 【如果 PD leader 未因 CPU 限制达到瓶颈而导致 TSO RPC 延迟升高】有歧义。这里是说 TSO RPC 延迟没有升高吗？

Author: 是不是说 TSO RPC 延迟升高了但是这个升高不是因为 PD leader CPU 达到瓶颈而导致的呢？

Author: 刚刚确认到是指： TSO gRPC 的延迟升高不是因为 PD leader CPU 达到瓶颈（是由网络等问题导致的）

Reviewer: @en-jin19 谢谢，能辛苦根据“TSO gRPC 的延迟升高不是因为 PD leader CPU 达到瓶颈（是由网络等问题导致的）”这一句修改下中文吗？🙏

> > Suppose that the CPU of the PD leader does not reach a bottleneck but the TSO RPC latency increases for other reasons (such as network issues). In this case, enabling the TSO Follower Proxy might increase the execution latency in TiDB and affect the QPS performance of the cluster.

Author: ```suggestion
> Suppose that the TSO RPC latency increases for reasons other than a CPU bottleneck of the PD leader (such as network issues). In this case, enabling the TSO Follower Proxy might increase the execution latency in TiDB and affect the QPS performance of the cluster.
```

- Range: `[0, 9223372036854775807]`
- This variable is used to limit the maximum number of requests TiDB can send to TiKV at the same time. 0 means no limit.

### tidb_tso_client_batch_max_wait_time <span class="version-mark">New in v5.3</span>

- Scope: GLOBAL
- Default value: `0`
- Range: `[0, 10]`
- Unit: Milliseconds

> - This variable is used to set the maximum waiting time for a batch operation when TiDB requests TSO from PD. The default value is `0`, which means no extra waiting is performed.

Reviewer: ```suggestion
- This variable is used to set the maximum waiting time for a batch operation when TiDB requests TSO from PD. The default value is `0`, which means no extra waiting time.
```

> - When obtaining TSO requests from PD, PD client, used by TiDB, collects as many TSO requests that are received at the same time as possible at one time. Then, PD client consolidates those requests into one RPC request and sends them to PD, thereby reducing the pressure on PD.

Reviewer: ```suggestion
- When obtaining TSO requests from PD each time, PD Client, used by TiDB, collects as many TSO requests received at the same time as possible. Then, PD Client merges the collected requests in batch into one RPC request and sends the request to PD. This helps reduce the pressure on PD.
```

> - After setting this value other than 0, TiDB waits for the maximum duration of this value before the end of each batch. This is to collect more TSO requests, thereby improving the batching effect.

Reviewer 2: ```suggestion
- After setting this value greater than 0, TiDB waits for the maximum duration of this value before the end of each batch. This is to collect more TSO requests, thereby improving the batching effect.
```

> - After setting this value greater than 0, TiDB waits for the maximum duration of this value before the end of each batch. This is to collect more TSO requests, thereby improving the batching effect.

Reviewer: ```suggestion
- After setting this variable to a value greater than `0`, TiDB waits for the maximum duration of this value before the end of each batch merge. This is to collect more TSO requests and improve the effect of batch operations.
```

- Scenarios for increasing the value of this variable:

>     * Due to the high pressure of TSO requests, the CPU of PD leaders reaches a bottleneck, which causes high latency of TSO RPC requests.

Author: ```suggestion
    * Due to the high pressure of TSO requests, the CPU of the PD leader reaches a bottleneck, which causes high latency of TSO RPC requests.
```

>     * Due to the high pressure of TSO requests, the CPU of PD leaders reaches a bottleneck, and this issue causes high latency of TSO RPC requests.

Reviewer 2: ```suggestion
    * Due to the high pressure of TSO requests, the CPU of PD leaders reaches a bottleneck, which causes high latency of TSO RPC requests.
```

>     * There are not many TiDB instances in the cluster, but every TiDB instance is under relatively high concurrency.

Reviewer: ```suggestion
    * There are not many TiDB instances in the cluster, but every TiDB instance is in high concurrency.
```

- When setting this variable, it is recommended to set it to a smaller value as much as possible.

> **Notes:**
>

> > Suppose the TSO RPC latency increases, but the CPU bottleneck of PD leaders does not cause this issue. In this case, increasing the value of `tidb_tso_client_batch_max_wait_time` might increase the latency of TiDB statement execution, thereby affecting the QPS performance of the cluster.

Reviewer: ditto

Author: ```suggestion
> Suppose that the CPU of the PD leader does not reach a bottleneck but the TSO RPC latency increases for other reasons (such as network issues). In this case, increasing the value of `tidb_tso_client_batch_max_wait_time` might increase the execution latency in TiDB and affect the QPS performance of the cluster.
```
